src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs

```diff
@@ -656,5 +653,71 @@ private static bool LogFailure(ValueStopwatch stopwatch)
             NameResolutionTelemetry.Log.AfterResolution(stopwatch, successful: false);
             return false;
         }
+
+        /// <summary>Mapping from key to current task in flight for that key.</summary>
+        private static readonly Dictionary<object, Task> s_tasks = new Dictionary<object, Task>();
```

Comment:
I wonder if we should set the capacity to 10

Reply:
That seems arbitrary :)  If this grows to be >= 10 and we incur two extra resizes (0 to 3, 3 to 7) over the lifetime of the process, meh.

Reply:
But but... OK 😄